src/NServiceBus.Newtonsoft.Json/JsonMessageSerializer.cs

```diff
@@ -26,10 +27,11 @@ class JsonMessageSerializer : IMessageSerializer
         {
             this.messageMapper = messageMapper;
 
-            settings ??= new JsonSerializerSettings
+            if (settings == null)
             {
-                TypeNameHandling = TypeNameHandling.Auto
-            };
+                settings = new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto };
+                log.Warn($"The default {nameof(JsonSerializerSettings)} for NServiceBus.Newtonsoft.Json use TypeNameHandling.Auto for backwards compatibility. This is a potential security vulnerability and it is recommended to use TypeNameHandling.None if possible. To disable this warning, provide a custom {nameof(JsonSerializerSettings)} instance to 'endpointConfiguration.UseSerialization<NewtonsoftSerializer>().Settings'.");
```

Comment:
Should be link to docs search?
```suggestion
                log.Warn($"The default {nameof(JsonSerializerSettings)} for NServiceBus.Newtonsoft.Json uses TypeNameHandling.Auto for backwards compatibility. This is a potential security vulnerability and it is recommended to use TypeNameHandling.None if possible. To disable this warning, provide a custom {nameof(JsonSerializerSettings)} instance to 'endpointConfiguration.UseSerialization<NewtonsoftSerializer>().Settings'. See https://docs.particular.net/search?q=newtonsoftjson+typenamehandling");
```

Reply:
I think we usually don't hardcode links in exception messages as this might be a problem in the long run.

Reply:
@timbussmann That is why it links to search and not a specific page. It occasionally happens, see https://github.com/search?l=C%23&q=org%3AParticular+https%3A%2F%2Fdocs.particular.net%2Fsearch&type=Code